engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -649,7 +649,7 @@ protected void serviceStart() throws Exception
       addSSLConfigResource(config);
       WebApp webApp = WebApps.$for("stram", StramAppContext.class, appContext, "ws").with(config).start(new StramWebApp(this.dnmgr));
       LOG.info("Started web service at port: " + webApp.port());
-      appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getHostName() + ":" + webApp.port();
+      appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getAddress().getCanonicalHostName() + ":" + webApp.port();
```

Comment:
I don't think that java uses /etc/hostname directly in getHostName(). As far as I can tell, it does forward and then reverse name lookup in case of getHostName().

+1 for the suggestion to use getCanonicalHostName() and fallback to getHostName() especially that it will be consistent with what RM is doing. 

Note that on my machine getCanonicalHostName returns ip address while getHostName() does the proper resolution:
```
hostname = vrozov.local
canonical = 192.168.128.32
localhost = vrozov.local/192.168.128.32
```